src/components/FileUpload/hooks.test.js

```diff
@@ -14,80 +12,87 @@ jest.mock('hooks/app', () => ({
   })),
 }));
 
-const state = mockUseKeyedState(stateKeys);
-
 describe('File Upload hooks', () => {
   describe('useUploadConfirmModalHooks', () => {
     const props = {
       file: { fileName: 'file.pdf', fileUrl: 'http://example.com' },
       closeHandler: jest.fn(),
       uploadHandler: jest.fn(),
     };
+
+    let setStateSpy;
+    const setValue = jest.fn();
+
     beforeEach(() => {
-      jest.clearAllMocks();
-      state.mock();
+      setStateSpy = jest.spyOn(React, 'useState').mockImplementation((value) => [value, setValue]);
     });
+
     afterEach(() => {
-      state.resetVals();
+      setStateSpy.mockRestore();
+      jest.clearAllMocks();
     });
 
     it('start with initial state', () => {
       useUploadConfirmModalHooks(props);
-      state.expectInitializedWith(stateKeys.description, '');
-      state.expectInitializedWith(stateKeys.shouldShowError, false);
+      expect(setStateSpy).toHaveBeenCalledWith('');
+      expect(setStateSpy).toHaveBeenCalledWith(false);
```

Comment:
It'd be great to add comments here similar to the ones added in `FileRenderer/hooks.test.js`.